src/snowflake/cli/_plugins/spcs/services/manager.py

```diff
@@ -133,12 +135,61 @@ def status(self, service_name: str) -> SnowflakeCursor:
         return self.execute_query(f"CALL SYSTEM$GET_SERVICE_STATUS('{service_name}')")
 
     def logs(
-        self, service_name: str, instance_id: str, container_name: str, num_lines: int
+        self,
+        service_name: str,
+        instance_id: str,
+        container_name: str,
+        num_lines: int,
+        since_timestamp: str = "",
+        include_timestamps: bool = False,
     ):
         return self.execute_query(
-            f"call SYSTEM$GET_SERVICE_LOGS('{service_name}', '{instance_id}', '{container_name}', {num_lines});"
+            f"call SYSTEM$GET_SERVICE_LOGS('{service_name}', '{instance_id}', '{container_name}', "
+            f"{num_lines}, False, '{since_timestamp}', {include_timestamps});"
         )
 
+    def stream_logs(
+        self,
+        service_name: str,
+        instance_id: str,
+        container_name: str,
+        num_lines: int,
+        since_timestamp: str,
+        interval_seconds: int,
+    ):
+        try:
+            prev_timestamp = since_timestamp
+            prev_log_records: List[str] = []
+
+            while True:
+                cursor = self.logs(
+                    service_name=service_name,
+                    instance_id=instance_id,
+                    container_name=container_name,
+                    num_lines=num_lines,
+                    since_timestamp=prev_timestamp,
+                    include_timestamps=True,
+                )
+                new_logs = cursor.fetchall()
+
+                if new_logs:
+                    log_block = new_logs[0][0]
+                    new_log_records = log_block.split("\n")
+                    new_log_records = [line for line in new_log_records if line.strip()]
+                    if new_log_records:
+                        dedup_log_records = _new_logs_only(
+                            prev_log_records, new_log_records
+                        )
+                        yield dedup_log_records
```

Comment:
related to my above comment about moving the `filter_log_timestamp` into here, what you probably want is something like
```suggestion
                        for log in dedup_log_records:
                            yield filter_log_timestamp(log, include_timestamps)